### ISSUE TITLE
[0.9.1] Stopped Serving POST Requests

### ISSUE THREAD

**Issue by esecules** [influxdb/influxdb#3473]
I have a single influx 0.9.1 instance running on top of 3.7 GB of data on a machine with 4GB of ram

Last night the server stopped ingesting data from POST requests. The server kept serving chronograf's GET requests. At the time of failure there were 2000-3000 points going in per minute with maximum latencies less than 2.5s, with most under 0.5s. This is for sure far from the most stress the system has encountered.

It was fixed be deleting and restarting the docker container it is running in.

**Comment by beckettsean** [comment]
@esecules There's not enough in the initial report for us to reproduce or understand what might have happened.

Can you share logs and some sample of your writes and queries? What errors were you getting on POSTs? How did you install InfluxDB?

**Comment by esecules** [comment]
My python influx client was getting request reset by peer after the 30 second timeout I set.

Here are the logs from influx around the time of failure
```
[http] 2015/07/27 08:34:04 10.244.36.0 - - [27/Jul/2015:08:34:04 +0000] POST /write?p=root&u=root HTTP/1.1 204 0 - python-requests/2.7.0 CPython/2.7.3 Linux/4.0.7-coreos-r2 3d8089ef-343a-11e5-8edf-000000000000 55.297108ms
[http] 2015/07/27 08:34:04 10.244.36.0 - - [27/Jul/2015:08:34:04 +0000] POST /write?p=root&u=root HTTP/1.1 204 0 - python-requests/2.7.0 CPython/2.7.3 Linux/4.0.7-coreos-r2 3d897e6f-343a-11e5-8ee0-000000000000 12.199078ms
[http] 2015/07/27 08:34:05 10.244.36.0 - - [27/Jul/2015:08:34:05 +0000] POST /write?p=root&u=root HTTP/1.1 204 0 - python-requests/2.7.0 CPython/2.7.3 Linux/4.0.7-coreos-r2 3d8c9fed-343a-11e5-8ee1-000000000000 61.933956ms
[http] 2015/07/27 08:34:05 10.244.36.0 - - [27/Jul/2015:08:34:05 +0000] POST /write?p=root&u=root HTTP/1.1 204 0 - python-requests/2.7.0 CPython/2.7.3 Linux/4.0.7-coreos-r2 3d9673c5-343a-11e5-8ee2-000000000000 12.557393ms
[http] 2015/07/27 08:34:05 10.244.36.0 - - [27/Jul/2015:08:34:05 +0000] POST /write?p=root&u=root HTTP/1.1 204 0 - python-requests/2.7.0 CPython/2.7.3 Linux/4.0.7-coreos-r2 3d98b46f-343a-11e5-8ee3-000000000000 10.254508ms
[http] 2015/07/27 08:34:05 10.244.36.0 - - [27/Jul/2015:08:34:05 +0000] POST /write?p=root&u=root HTTP/1.1 204 0 - python-requests/2.7.0 CPython/2.7.3 Linux/4.0.7-coreos-r2 3d9a9630-343a-11e5-8ee4-000000000000 14.076772ms
[http] 2015/07/27 08:34:07 10.244.36.0 - - [27/Jul/2015:08:34:07 +0000] GET /query?q=SELECT%20SUM(%22value%22)%20FROM%20%22database%22..%22cluster.vms%22%20WHERE%20time%20%3E%20now()%20-%2024h%20GROUP%20BY%20time(20m) HTTP/1.1 200 4
63 http://10.128.63.123:58131/ Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:26.0) Gecko/20100101 Firefox/26.0 3f01ec15-343a-11e5-8ee5-000000000000 14.458911ms
[http] 2015/07/27 08:34:09 10.244.36.0 - - [27/Jul/2015:08:34:09 +0000] GET /query?q=SELECT%20mean(%22value%22)%20FROM%20%22database%22..%22cluster.latency.read%22%20WHERE%20time%20%3E%20now()%20-%2024h%20GROUP%20BY%20time(10m)%3B%2
0SELECT%20mean(%22value%22)%20FROM%20%22database%22..%22cluster.latency.write%22%20WHERE%20time%20%3E%20now()%20-%2024h%20GROUP%20BY%20time(10m) HTTP/1.1 200 4092 http://10.128.63.123:58131/ Mozilla/5.0 (X11; Ubuntu; Linux x86_64; r
v:26.0) Gecko/20100101 Firefox/26.0 407d8724-343a-11e5-8ee6-000000000000 13.951463ms
[http] 2015/07/27 08:34:11 10.244.36.0 - - [27/Jul/2015:08:34:10 +0000] GET /query?q=SELECT%20count(%22value%22)%20FROM%20%22database%22..%22events%22%20WHERE%20time%20%3E%20now()%20-%202h%20GROUP%20BY%20time(1m)%2C%20%22type%22%2C%
20%22serial%22 HTTP/1.1 200 3422 http://10.128.63.123:58131/ Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:26.0) Gecko/20100101 Firefox/26.0 408a3aaa-343a-11e5-8ee7-000000000000 1.446190671s
[http] 2015/07/27 08:34:26 10.244.36.0 - - [27/Jul/2015:08:34:24 +0000] GET /query?q=SELECT%20count(%22value%22)%20FROM%20%22database%22..%22events%22%20WHERE%20time%20%3E%20now()%20-%206h%20GROUP%20BY%20time(5m)%2C%20%22type%22%20f
ill(0) HTTP/1.1 200 2312 http://10.128.63.123:58131/ Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:26.0) Gecko/20100101 Firefox/26.0 492d9ce0-343a-11e5-8ee9-000000000000 2.44159332s
[retention] 2015/07/27 08:37:07 retention policy enforcement check commencing
[retention] 2015/07/27 08:37:07 retention policy shard deletion check commencing
[http] 2015/07/27 08:34:29 10.244.36.0 - - [27/Jul/2015:08:34:29 +0000] GET /query?q=SELECT%20max(%22value%22)%20FROM%20%22database%22..%22influx.latency.write%22%20WHERE%20time%20%3E%20now()%20-%201h%20GROUP%20BY%20time(1m)%3B%20SE
LECT%20mean(%22value%22)%20FROM%20%22database%22..%22influx.latency.write%22%20WHERE%20time%20%3E%20now()%20-%201h%20GROUP%20BY%20time(1m)%3B%20SELECT%20min(%22value%22)%20FROM%20%22database%22..%22influx.latency.write%22%20WHERE%20
time%20%3E%20now()%20-%201h%20GROUP%20BY%20time(1m) HTTP/1.1 200 2259 http://10.128.63.123:58131/ Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:26.0) Gecko/20100101 Firefox/26.0 4c15618f-343a-11e5-8eea-000000000000 7.812307ms
```

**Comment by esecules** [comment]
Could this have something to do with the python client version Im using?

**Comment by beckettsean** [comment]
@esecules given that there's Docker and the python client involved, there's too much for us to try and repro easily. I don't see anything indicative of a failure, and the logs don't show any 4xx or 5xx errors on POST. Can you clarify what you mean by "server stopped ingesting data from POST requests"? From the logs you've given it looks like the POST requests stopped making it to InfluxDB entirely.

**Comment by esecules** [comment]
By "server stopped ingesting data from POST requests" i meant that from 2015/07/27 08:34:07 onward no more post requests showed up in the logs and the python client saw 104 errors.

I have not been able to repro either (partially because i have been tied up with #3493 and #3492), but I also think it could be the python client. However I am hesitant to believe it was a Kubernetes pod networking issue since chronograf (also in a Kubernetes pod)  could send GET requests to Influx.

**Comment by beckettsean** [comment]
I've not seen any other instances of the write endpoint disappearing like that, and it's clear from the InfluxDB logs that the POST requests weren't making it through at all. It's possible the `/write` endpoint died somehow. The `GET` requests from chronograf go to the `/query` endpoint. Were both still configured on port 8086?

I notice from this and your other issues that you're running 0.9.1. I would recommend upgrading to 0.9.2 for a number of stability and performance tweaks. It's a drop-in replacement upgrade.

**Comment by esecules** [comment]
I do have both endpoints hooked to port 8086 and I will upgrade to 0.9.2 soon

**Comment by esecules** [comment]
I have managed to reproduce this!

**Comment by beckettsean** [comment]
A 500 error from the write endpoint indicates a fatal timeout, but once the deadlock or performance issue abates it should be possible to write again. Does it eventually heal or is the write endpoint returning 500s for more than 5-10 minutes?

**Comment by esecules** [comment]
sorry for the confusion I realized my mistake created a new issue then realized it created it then found its duplicate

**Comment by dominikh** [comment]
I've encountered the same problem, under very light load (writing a dozen points a second, selecting 600 points every 5 seconds). Almost all POSTs timed out after 5 seconds, returning a 500, but some succeeded with 204. This kept happening for 20 or so minutes. The server itself was under light load only, too. influxdb wasn't consuming much CPU.

```
[http] 2015/08/09 20:46:38 127.0.0.1 - - [09/Aug/2015:20:46:33 +0200] POST /write?consistency=&db=router&precision=&rp=hour HTTP/1.1 500 32 - InfluxDBClient f4d010c4-3ec6-11e5-8ec3-000000000000 5.001281822s
[http] 2015/08/09 20:46:43 127.0.0.1 - - [09/Aug/2015:20:46:38 +0200] POST /write?consistency=&db=router&precision=&rp=hour HTTP/1.1 500 32 - InfluxDBClient f7cb609f-3ec6-11e5-8ec4-000000000000 5.00136886s
[http] 2015/08/09 20:46:48 127.0.0.1 - - [09/Aug/2015:20:46:43 +0200] POST /write?consistency=&db=router&precision=&rp=hour HTTP/1.1 500 32 - InfluxDBClient fac6e909-3ec6-11e5-8ec5-000000000000 5.001399751s
[http] 2015/08/09 20:46:48 127.0.0.1 - - [09/Aug/2015:20:46:48 +0200] POST /write?consistency=&db=router&precision=&rp=hour HTTP/1.1 204 0 - InfluxDBClient fdc234ed-3ec6-11e5-8ec7-000000000000 161.924µs
[http] 2015/08/09 20:46:54 127.0.0.1 - - [09/Aug/2015:20:46:49 +0200] POST /write?consistency=&db=router&precision=&rp=hour HTTP/1.1 500 32 - InfluxDBClient fe590f0a-3ec6-11e5-8ec8-000000000000 5.01703416s
```

Some GET requests, like SELECT or SHOW MEASUREMENTS hang completely and indefinitely, and don't even show up in the log. SHOW DATABASES, however, succeeds and shows up in the log.

In the end, I SIGQUIT the process to get a stacktrace. Unfortunately, influxdb was running in a tmux with insufficient (~5000) lines of scrollback, so the output is truncated, but probably still pinpoints the problem: http://sprunge.us/YfMD - I don't know much about influxdb's internals, but lots of 20 minute locks trying to write don't seem kosher to me.

This is on InfluxDB v0.9.2 (git: 6c0a91f775f9fc0e625d17ffa04a3fe86945ba09)

**Comment by dominikh** [comment]
More log output (of a new incident, this time the moment it started) suggests that this is related to continuous queries, so related/a duplicate of all the other open issues regarding CQs and timeouts:

```
[http] 2015/08/10 11:45:14 127.0.0.1 - - [10/Aug/2015:11:45:14 +0200] POST /write?consistency=&db=router&precision=&rp=hour HTTP/1.1 204 0 - InfluxDBClient 8049436c-3f44-11e5-8035-000000000000 14.052909ms
[http] 2015/08/10 11:45:20 127.0.0.1 - - [10/Aug/2015:11:45:15 +0200] POST /write?consistency=&db=router&precision=&rp=hour HTTP/1.1 500 32 - InfluxDBClient 80e1ce96-3f44-11e5-8036-000000000000 5.010775773s
[continuous_querier] 2015/08/10 11:45:21 timeout
[continuous_querier] 2015/08/10 11:45:21 error: timeout. running: SELECT mean(value) INTO "router"."day"."traffic.in.avg" FROM "router"."hour"."traffic.in" WHERE time >= '2015-08-10T09:45:00Z' AND time < '2015-08-10T09:45:30Z' GROUP BY time(30s), *
[continuous_querier] 2015/08/10 11:45:21 error executing query: CREATE CONTINUOUS QUERY traffic_in_daily_avg ON router BEGIN SELECT mean(value) INTO "router"."day"."traffic.in.avg" FROM "router"."hour"."traffic.in" GROUP BY time(30s), * END: err = timeout
[http] 2015/08/10 11:45:25 127.0.0.1 - - [10/Aug/2015:11:45:20 +0200] POST /write?consistency=&db=router&precision=&rp=hour HTTP/1.1 500 32 - InfluxDBClient 83deaa00-3f44-11e5-8037-000000000000 5.001440302s
[http] 2015/08/10 11:45:30 127.0.0.1 - - [10/Aug/2015:11:45:25 +0200] POST /write?consistency=&db=router&precision=&rp=hour HTTP/1.1 500 32 - InfluxDBClient 86da0b55-3f44-11e5-8038-000000000000 5.0078704s
[http] 2015/08/10 11:45:30 127.0.0.1 - - [10/Aug/2015:11:45:30 +0200] POST /write?consistency=&db=router&precision=&rp=hour HTTP/1.1 204 0 - InfluxDBClient 89d66a96-3f44-11e5-8039-000000000000 141.994µs
[http] 2015/08/10 11:45:36 127.0.0.1 - - [10/Aug/2015:11:45:31 +0200] POST /write?consistency=&db=router&precision=&rp=hour HTTP/1.1 500 32 - InfluxDBClient 8a6b6d0c-3f44-11e5-803a-000000000000 5.002164044s
[http] 2015/08/10 11:45:41 127.0.0.1 - - [10/Aug/2015:11:45:36 +0200] POST /write?consistency=&db=router&precision=&rp=hour HTTP/1.1 500 32 - InfluxDBClient 8d66f515-3f44-11e5-803b-000000000000 5.009831064s
[http] 2015/08/10 11:45:41 127.0.0.1 - - [10/Aug/2015:11:45:41 +0200] POST /write?consistency=&db=router&precision=&rp=hour HTTP/1.1 204 0 - InfluxDBClient 906395f7-3f44-11e5-803c-000000000000 142.984µs
[http] 2015/08/10 11:45:47 127.0.0.1 - - [10/Aug/2015:11:45:42 +0200] POST /write?consistency=&db=router&precision=&rp=hour HTTP/1.1 500 32 - InfluxDBClient 90f9dff0-3f44-11e5-803d-000000000000 5.00231094s
[http] 2015/08/10 11:45:47 127.0.0.1 - - [10/Aug/2015:11:45:47 +0200] POST /write?consistency=&db=router&precision=&rp=hour HTTP/1.1 204 0 - InfluxDBClient 93f59ac4-3f44-11e5-803e-000000000000 223.012µs
^C[run] 2015/08/10 11:45:50 signal received, initializing clean shutdown...
[run] 2015/08/10 11:45:50 waiting for clean shutdown...
[tcp] 2015/08/10 11:45:50 cluster service accept error: network connection closed
[snapshot] 2015/08/10 11:45:50 snapshot listener closed
[http] 2015/08/10 11:45:53 127.0.0.1 - - [10/Aug/2015:11:45:48 +0200] POST /write?consistency=&db=router&precision=&rp=hour HTTP/1.1 500 32 - InfluxDBClient 948d818d-3f44-11e5-803f-000000000000 5.010668518s
[http] 2015/08/10 11:45:53 127.0.0.1 - - [10/Aug/2015:11:45:53 +0200] POST /write?consistency=&db=router&precision=&rp=hour HTTP/1.1 204 0 - InfluxDBClient 978a3def-3f44-11e5-8040-000000000000 185.37µs
[http] 2015/08/10 11:45:59 127.0.0.1 - - [10/Aug/2015:11:45:54 +0200] POST /write?consistency=&db=router&precision=&rp=hour HTTP/1.1 500 32 - InfluxDBClient 9820e85d-3f44-11e5-8041-000000000000 5.001971407s
[http] 2015/08/10 11:45:59 127.0.0.1 - - [10/Aug/2015:11:45:59 +0200] POST /write?consistency=&db=router&precision=&rp=hour HTTP/1.1 204 0 - InfluxDBClient 9b1c5a2a-3f44-11e5-8042-000000000000 142.572µs
^C[run] 2015/08/10 11:46:02 second signal received, initializing hard shutdown

```

**Comment by esecules** [comment]
@dominikh I dont think this is the same issue. It looks more like #3411.

**Comment by beckettsean** [comment]
@dominikh I agree, I think your issue has been addressed with the write timeout tweaks.

@esecules are you seeing this recur with 0.9.2+?

**Comment by beckettsean** [comment]
Should be solved with 0.9.3. Please reopen if that isn't the case.
